app/models/attachment.rb

```diff
@@ -11,4 +13,16 @@ class Attachment < ActiveRecord::Base
   def name
     file_file_name
   end
+
+  def expiring_url
+    file.expiring_url(Time.now + EXPIRE_TIMEFRAME)
+  end
+
+  def course_department
+    self.attachable.course_department
```

Comment:
I don't think you need the `self` on these. 

You could also do this with a `delegates` call, I think:

```
delegates :course_department, :subject_department, to: :attachable
```

That's what we've done in other places. Having said that, I think I might actually prefer writing the methods out as you have done here. It seems more clear. What do you think?